### PR TITLE
Source Posgres: [CDC] allow null values for columns

### LINKED ISSUE #8557: 🐛 Source Postgres (CDC):  Sync fails when records in Postgres are deleted for tables with non null constraint on columns
## Environment
- **Airbyte version**:  v.30.39 **(upgraded to v.0.32.6)**
- **OS Version / Instance**: debian / GCP e2-medium
- **Deployment**: Docker
- **Source Connector and version**: Postgres 0.3.13 **(upgraded to 0.3.17)**
- **Destination Connector and version**: Bigquery 0.5.0
- **Severity**: High
- **Step where error happened**: Sync job

## Current Behavior
Trying to setup CDC with postgres 12 (CloudSQL hosted)-> Bigquery, keep getting errors when data is deleted. Think it has something to do with that the postgres schema has non null constraints on certain columns. The deletes are picked up as row [id, null, null etc..] and since some of the columns have non null constraints it produces an error. Have tried with both wall2json and pgoutput

## Expected Behavior
Expected behavior is that the delete events should be picked up by debezium and not throw any errors, since the non null constraints are only relevant when writing to the Postgres table, not reading from it.

## Logs

<details>
<summary>LOG</summary> 

```

[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-12 14:15:01 [1;31mERROR[m i.d.r.TableSchemaBuilder(lambda$createValueGenerator$5):269 - {} - Failed to properly convert data value for 'public.countries.code' of type varchar for row [090add40-0a14-11eb-b226-6f1fbcfccacc, null, null, null, null, null]:
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - org.apache.kafka.connect.errors.DataException: Invalid value: null used for required field: "code", schema type: STRING
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at org.apache.kafka.connect.data.ConnectSchema.validateValue(ConnectSchema.java:220) ~[connect-api-2.6.1.jar:?]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at org.apache.kafka.connect.data.Struct.put(Struct.java:216) ~[connect-api-2.6.1.jar:?]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.relational.TableSchemaBuilder.lambda$createValueGenerator$5(TableSchemaBuilder.java:265) ~[debezium-core-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.relational.TableSchema.valueFromColumnData(TableSchema.java:141) ~[debezium-core-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.relational.RelationalChangeRecordEmitter.emitDeleteRecord(RelationalChangeRecordEmitter.java:130) ~[debezium-core-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.relational.RelationalChangeRecordEmitter.emitChangeRecords(RelationalChangeRecordEmitter.java:54) ~[debezium-core-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.connector.postgresql.PostgresChangeRecordEmitter.emitChangeRecords(PostgresChangeRecordEmitter.java:88) ~[debezium-connector-postgres-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.pipeline.EventDispatcher.dispatchDataChangeEvent(EventDispatcher.java:198) ~[debezium-core-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.lambda$processMessages$0(PostgresStreamingChangeEventSource.java:226) ~[debezium-connector-postgres-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.connector.postgresql.connection.wal2json.NonStreamingWal2JsonMessageDecoder.processNotEmptyMessage(NonStreamingWal2JsonMessageDecoder.java:78) ~[debezium-connector-postgres-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.connector.postgresql.connection.AbstractMessageDecoder.processMessage(AbstractMessageDecoder.java:42) ~[debezium-connector-postgres-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.deserializeMessages(PostgresReplicationConnection.java:482) ~[debezium-connector-postgres-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.readPending(PostgresReplicationConnection.java:474) ~[debezium-connector-postgres-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.processMessages(PostgresStreamingChangeEventSource.java:185) ~[debezium-connector-postgres-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.execute(PostgresStreamingChangeEventSource.java:147) ~[debezium-connector-postgres-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.pipeline.ChangeEventSourceCoordinator.streamEvents(ChangeEventSourceCoordinator.java:140) ~[debezium-core-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at io.debezium.pipeline.ChangeEventSourceCoordinator.lambda$start$0(ChangeEventSourceCoordinator.java:113) ~[debezium-core-1.4.2.Final.jar:1.4.2.Final]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
[34msource[0m - 2021-11-12 14:15:01 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 	at java.lang.Thread.run(Thread.java:832) [?:?]

```

</details>

## Steps to Reproduce
1. Set up CDC replication sync from postgres 12 to bigquery
2. Run a sync
3. Delete a row from the postgres source table 
4. Run another sync

## Are you willing to submit a PR?
Do not know how to solve this, it seems like and issue with debezium

### PR DESCRIPTION
**Pull request by yurii-bidiuk**
## What
Fixes #8557 

## How
Added 'null' values to shema building.
**Before** schema for column looked like `{ "type": "string" }`
**Now**: `{ "type": [ "string", null ] }`

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
There is no user impact

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by alexandertsukanov** [comment]
LGTM

**Review by alexandertsukanov — APPROVED** [review]

File: airbyte-integrations/connectors/source-postgres/Dockerfile [hunk 1]
```diff
@@ -16,5 +16,5 @@ ENV APPLICATION source-postgres
 
 COPY --from=build /airbyte /airbyte
 
-LABEL io.airbyte.version=0.4.16
+LABEL io.airbyte.version=0.4.17
```
Comment (alexandr-shegeda) [inline airbyte-integrations/connectors/source-postgres/Dockerfile:19]: please checkout latest master

**Review by alexandr-shegeda — APPROVED** [review]
LGTM. we can proceed with publishing and merging without airbyte review

File: airbyte-integrations/connectors/source-postgres/Dockerfile (hunk 1 above)
Comment (yurii-bidiuk) [reply airbyte-integrations/connectors/source-postgres/Dockerfile:19]: the last version is 0.4.16

**Comment by grishick** [comment]
If I understand the changes correctly, this will make all Date, Text, and Money fields nullable (optional). I don't think that's the correct change though, because it will also allow non-nullable columns to be NULLs. Could we use `RelationalColumn::isOptional` to detect if a column is nullable?

**Comment by yurii-bidiuk** [comment]
@grishick 
I tested these changes and it doesn't break any logic now. If I add optional only for nullable fields it will produce #8557. The reason is that delete event produces a message with all fields set as null [1, null, null, null] and if some of these fields has NOT NULL constrain, we will get #8557

**Comment by grishick** [comment]
/publish connector=connectors/source-postgres run-tests=false

**Comment by grishick** [comment]
I reran the build and tests locally to make sure they pass.

**Comment by yurii-bidiuk** [comment]
@grishick thank you for helping with this issue

**Pull request merged**